### PR TITLE
testing: use test dialect and custom passes for test_get_all_available_passes

### PR DESCRIPTION
**Pull request by superlopuh**
No more updating the pass list when we add a new transform!

### CODE REVIEW

File: tests/interactive/test_get_all_available_passes.py [hunk 1]
```diff
@@ -1,65 +1,111 @@
...
+        },
     )
 
-    expected_res = tuple(
+    assert res == tuple(
         (
             AvailablePass(
-                display_name="reconcile-unrealized-casts",
-                module_pass=reconcile_unrealized_casts.ReconcileUnrealizedCastsPass,
+                display_name="bc",
```
Comment (n-io) [inline tests/interactive/test_get_all_available_passes.py:89]:
```suggestion
                display_name=BCPass.name,
```

File: tests/interactive/test_get_all_available_passes.py [hunk 2]
```diff
@@ -1,65 +1,111 @@
...
+                module_pass=BCPass,
                 pass_spec=None,
             ),
             AvailablePass(
-                display_name="test-lower-linalg-to-snitch",
-                module_pass=test_lower_linalg_to_snitch.TestLowerLinalgToSnitchPass,
+                display_name="bd",
```
Comment (n-io) [inline tests/interactive/test_get_all_available_passes.py:94]:
```suggestion
                display_name=BDPass.name,
```

**Review by n-io — APPROVED** [review]
Looks good, I think this is a solid way of testing. Feel free to ignore my very minor code suggestions.

**Review by webmiche — COMMENTED** [review]
the works lord!

**Review by dshaaban01 — COMMENTED** [review]
👍👍

**Pull request merged**